src/Identity/UI/src/IdentityBuilderUIExtensions.cs

```diff
@@ -36,5 +63,90 @@ public static IdentityBuilder AddDefaultUI(this IdentityBuilder builder)
 
             return builder;
         }
+
+        private static Assembly GetApplicationAssembly(IdentityBuilder builder)
+        {
+            // Whis is the same logic that MVC follows to find the application assembly.
+            var environment = builder.Services.Where(d => d.ServiceType == typeof(IWebHostEnvironment)).ToArray();
+            var applicationName = ((IWebHostEnvironment)environment.LastOrDefault()?.ImplementationInstance)
+                .ApplicationName;
+
+            if (applicationName == null)
+            {
+                return null;
+            }
+            var appAssembly = Assembly.Load(applicationName);
+            return appAssembly;
+        }
+
+        private static bool TryResolveUIFramework(Assembly assembly, out UIFramework uiFramework)
+        {
+            uiFramework = default;
+
+            var metadata = assembly?.GetCustomAttributes<UIFrameworkAttribute>()
+                .SingleOrDefault()?.UIFramework; // Bootstrap5 is the default
+            if (metadata == null)
+            {
+                return false;
+            }
+
+            // If we find the metadata there must be a valid framework here.
+            if (!Enum.TryParse(metadata, ignoreCase: true, out uiFramework))
+            {
+                var enumValues = string.Join(", ", Enum.GetNames(typeof(UIFramework)).Select(v => $"'{v}'"));
+                throw new InvalidOperationException(
+                    $"Found an invalid value for the 'IdentityUIFrameworkVersion'. Valid values are {enumValues}");
+            }
+
+            return true;
+        }
+
+        internal class ViewVersionFeatureProvider : IApplicationFeatureProvider<ViewsFeature>
+        {
+            private readonly bool _isV4;
```

Comment:
I see a few places in the code where there's an assumption that v4 and v5 are the only versions. In the interest of the next time we have to do this (let's say it's v5 and v6 SxS at some point) I think it's worth spending a bit of effort making things fail in obvious ways. For instance, today, if _isV4 is false, we always take the v5 path... if we change that to use an enum for the version instead, and fail loudly if the version is neither 4 nor 5, it will be way easier the next time around since we'll know what needs to be changed.

Reply:
Sure we already have a version enum that we can use, I'll switch to harden this logic

Reply:
Note, with the new logic, we might not need any follow up validation for the versions as we get the following error at Startup time now with the enum check

```
Unhandled exception. System.InvalidOperationException: Found an invalid value for the 'IdentityUIFrameworkVersion'. Valid values are 'Bootstrap5', 'Bootstrap4'
```